cmd/xl-v1-utils.go

```diff
@@ -343,3 +343,24 @@ func getOrderedDisks(distribution []int, disks []StorageAPI) (orderedDisks []Sto
 	}
 	return orderedDisks
 }
+
+// getPartSizeFromIdx predicts the part size according to its index. It also
+// returns -1 when totalSize is also -1.
+func getPartSizeFromIdx(totalSize int64, partSize int64, partIndex int) int64 {
```

Comment:
Done, also I've added some panic() inside getPartSizeFromIdx for unmeaningful parameters like negative part index